src/model/TaskManager.cpp

```diff
@@ -48,76 +84,85 @@ std::vector<Core::TaskEntity> TaskManager::getTasks() const {
 std::vector<Core::TaskEntity> TaskManager::getTasks(const std::string &label) const {
     std::vector<Core::TaskEntity> tasks;
     for (const auto &kv : tasks_) {
-        if (kv.second.first.label() == label) {
-            Core::TaskEntity te;
-            te.set_allocated_id(new Core::TaskID(kv.first));
-            te.set_allocated_data(new Core::Task(kv.second.first));
-            tasks.push_back(te);
+        for (const auto &task_label : kv.second.first.labels()) {
+            if (task_label == label) {
+                Core::TaskEntity te;
+                te.set_allocated_id(new Core::TaskID(kv.first));
+                te.set_allocated_data(new Core::Task(kv.second.first));
+                tasks.push_back(te);
```

Comment:
consider `emplace_back`

Reply:
Will do for the future. Here, TaskEntity does not have a id+data constructor (and I want the parent field to stay unassigned)

Reply:
ultimately, used `emplace_back` with copy constructor and factory method